## Dehli Indian Housing Exploratory Data Analysis by Dumisani Maxwell Mukuchura

#### Dataset Source: *https://www.kaggle.com/datasets/bhavyadhingra00020/india-rental-house-price*

##### Contact: dumisanimukuchura@gmail.com LinkedIn: https://www.linkedin.com/in/dumisani-maxwell-mukuchura-4859b7170/

##### This Project will explore the Dehli Indian Housing and understand what factors are influential to price

In [1]:
#Import Standard Modules and Libraies to use

import os #For file and directory operations

#Data Libraries

import pandas as pd
import numpy as np

#Visualization Libraries
import matplotlib.pyplot as plt
import plotly.express as px

### Defining file path and import 'Indian_housing_Delhi_data.csv'

In [15]:
#Defining file path and import 'Indian_housing_Delhi_data.csv'
# Get the current working directory
current_dir = os.getcwd()

# Trim to one level above
base_dir = os.path.dirname(current_dir)

# Construct the path to the data folder
data_dir = os.path.join(base_dir, "data")

# Construct the full path to the CSV file
csv_file_path = os.path.join(data_dir, "Indian_housing_Delhi_data.csv")

# Read the CSV file into a DataFrame
housing_data = pd.read_csv(csv_file_path)

#Make a Copy and Maintain the original dataset as is 
housing_df = housing_data.copy()

#Check if import was successful with a head() check
housing_data.head()

house_type   house_size           location   city   latitude  \
0    1 RK Studio Apartment     400 sq ft            Kalkaji  Delhi  28.545561   
1    1 RK Studio Apartment     400 sq ft  Mansarover Garden  Delhi  28.643259   
2  2 BHK Independent Floor     500 sq ft        Uttam Nagar  Delhi  28.618677   
3  3 BHK Independent House   1,020 sq ft         Model Town  Delhi  28.712898   
4          2 BHK Apartment     810 sq ft   Sector 13 Rohini  Delhi  28.723539   

   longitude  price currency  numBathrooms  numBalconies isNegotiable  \
0  77.254349  22000      INR           1.0           NaN          NaN   
1  77.132828  20000      INR           1.0           NaN          NaN   
2  77.053352   8500      INR           1.0           NaN          NaN   
3  77.180000  48000      INR           3.0           NaN          NaN   
4  77.131424  20000      INR           2.0           NaN          NaN   

   priceSqFt    verificationDate  \
0        NaN    Posted a day ago   
1        NaN   Posted 9 days ago   
2        NaN  Posted 12 days ago   
3        NaN   Posted a year ago   
4        NaN   Posted a year ago   

                                         description SecurityDeposit  \
0  Fully furnished, loaded with amenities & gadge...      No Deposit   
1  Here is an excellent 1 BHK Independent Floor a...      No Deposit   
2  Zero Brokerage.\n\n2 Room set, Govt bijali Met...      No Deposit   
3  It’s a 3 bhk independent house situated in M...      No Deposit   
4  Well designed 2 bhk multistorey apartment is a...      No Deposit   

           Status  
0       Furnished  
1       Furnished  
2  Semi-Furnished  
3       Furnished  
4     Unfurnished

### Basic Exploration to understand the Dataset

In [16]:
#Data types and non-null values

housing_df.info() 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5000 entries, 0 to 4999
Data columns (total 16 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   house_type        5000 non-null   object 
 1   house_size        5000 non-null   object 
 2   location          5000 non-null   object 
 3   city              5000 non-null   object 
 4   latitude          5000 non-null   float64
 5   longitude         5000 non-null   float64
 6   price             5000 non-null   int64  
 7   currency          5000 non-null   object 
 8   numBathrooms      4975 non-null   float64
 9   numBalconies      2737 non-null   float64
 10  isNegotiable      179 non-null    object 
 11  priceSqFt         0 non-null      float64
 12  verificationDate  5000 non-null   object 
 13  description       4715 non-null   object 
 14  SecurityDeposit   5000 non-null   object 
 15  Status            5000 non-null   object 
dtypes: float64(5), int64(1), object(10)
memory

In [10]:
#Dimesnions of the dDataFrame

housing_df.shape

(5000, 16)

In [11]:
#Statistical Summary
housing_df.describe()

latitude    longitude         price  numBathrooms  numBalconies  \
count  5000.000000  5000.000000  5.000000e+03   4975.000000   2737.000000   
mean     28.578012    77.174499  2.221738e+05      2.918593      1.954330   
std       0.190186     0.115636  2.739843e+05      1.087823      0.547219   
min      20.011379    72.771332  3.000000e+03      1.000000      1.000000   
25%      28.544489    77.138248  2.950000e+04      2.000000      2.000000   
50%      28.569295    77.196472  1.250000e+05      3.000000      2.000000   
75%      28.618687    77.228950  3.011020e+05      4.000000      2.000000   
max      28.805466    80.361313  3.010101e+06     10.000000      8.000000   

       priceSqFt  
count        0.0  
mean         NaN  
std          NaN  
min          NaN  
25%          NaN  
50%          NaN  
75%          NaN  
max          NaN

In [12]:
#Check for missing  Values 

housing_df.isnull().sum()

house_type             0
house_size             0
location               0
city                   0
latitude               0
longitude              0
price                  0
currency               0
numBathrooms          25
numBalconies        2263
isNegotiable        4821
priceSqFt           5000
verificationDate       0
description          285
SecurityDeposit        0
Status                 0
dtype: int64

In [14]:
#Identifying unique value counts for each column 

housing_df.nunique()

house_type            28
house_size           339
location             288
city                   1
latitude             993
longitude            962
price                709
currency               1
numBathrooms          10
numBalconies           7
isNegotiable           1
priceSqFt              0
verificationDate      52
description         3820
SecurityDeposit      646
Status                 3
dtype: int64

### Thoughts from initial basic exploration of DataFrame 'housing_df'

- Dataframe has 16 columns and 5000 rows, 0 to 4999
- Shape of DataFrame is (5000, 16)
- 5 columns have missing values/ NaN values: 'numBathrooms', 'numBalconies', 'isNegotiable', 'priceSqFt', 'description'